src/msgraph_core/requests/batch_request_builder.py

```diff
@@ -108,6 +108,11 @@ async def _post_batch_collection(
 
         for batch_request_content in batch_request_content_collection.batches:
             request_info = await self.to_post_request_information(batch_request_content)
```

Comment:
I think we're crossing boundaries here in terms of responsibilities.
If we look at [the dotnet implementation](https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/2c3168974aeac7d83a335dcac484cc8425d6d186/src/Microsoft.Graph.Core/Requests/BatchRequestBuilder.cs#L77), the batch request collection is responsible for providing a well formed request.
Let's align on that implementation.

Reply:
You mean given a collection, we don't need to break it down to its content? the collection to prepare the request without breaking it down to a request collection content?

Reply:
if by "breaking down" you mean splitting the collection in groups of 20, no, I'm not suggesting we change that logic. However, I'm suggesting we move the "wrapping" of the requests with a parent object to the collection.
Again, take the time to read the dotnet implementation, it should illustrate my comments pretty well.

Reply:
To clarify further what @baywet is saying, when the `to_post_request_information` is called in the line above, it will invoke the serializer to set the payload. Since the model is already self-describing at the link below, the `requests` property should already be set properly at this time.
https://github.com/microsoftgraph/msgraph-sdk-python-core/blob/e22bff3dcc59e0ab395a75577ca33632ad044f27/src/msgraph_core/requests/batch_request_content.py#L140

Reply:
I'm not sure why this comment has been resolved since the underlying issue has not been addressed?
@musale can you please take this over since Shem is going to be out for a while?

Reply:
I'll look at this